app/js/components/tabs/tabcomponents/patientComponent.jsx

```diff
@@ -49,12 +49,23 @@ class PatientComponent extends Component {
 
   searchDemographics(event) {
     event.preventDefault();
-    const { gender, minAge, maxAge, livingStatus } = this.state;
-    let { startDate, endDate } = this.state;
+    const { gender, livingStatus } = this.state;
+    let { startDate, endDate, minAge, maxAge } = this.state;
     const searchParameters = { gender };
     // add appropriate age constraints to the search parameters
     if (minAge || maxAge) {
       if (minAge && maxAge) {
+        // coerce the values of minAge and maxAge to numbers for comparison    
+        minAge = Number(minAge);
+        maxAge = Number(maxAge);
+
+        // switch the min and max ages if the min age is greater than the max age
+        if (minAge > maxAge) {
+          const minAgeStore = minAge;
```

Comment:
- @kingisaac95 The user input can be controlled from the text-box, we do so via age validation. I and Paul Upendo, worked on a related issue, we added age validation to the age input field, more details can be found [Link to issue](https://issues.openmrs.org/browse/CB-97?jql=project%20%3D%20CB%20and%20status%20!%3D%20closed). However we did not implement this fix in our PR because it infringed upon the specs of this ticket, I would recommend you take a look at Paul [Pull Request](https://github.com/openmrs/openmrs-owa-cohortbuilder/pull/98/files) and build on it. Thanks.